src/settings/SettingRegistry.cpp

```diff
@@ -121,7 +121,7 @@ bool SettingRegistry::getDefinitionFile(const std::string machine_id, const std:
         char* path = strtok(paths,";:,"); // search for path delimited by ';', ':', ','
         while (path != NULL)
         {
-            result = std::string(path) + std::string("/") + machine_id + std::string(".def.json");
+            result = std::string(path) + std::string("/") + machine_id;
```

Comment:
I don't understand this commit. In Cura 2.2 the `inherit`field of a json file doesn't include the extension, so it must be added here to get the full file name.